core/components/header/schema.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from 'zod';
+
+import type { CurrencyCode } from './fragment';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
Feedback welcome, I tried many approaches to avoid a type assertion but I think it's actually safer than alternatives

Reply:
You should be able to remove this since you are doing `val is CurrencyCode` below.